src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -39,104 +39,22 @@ namespace app {
 namespace Clusters {
 namespace ModeBase {
 
-void Instance::RegisterInstance()
-{
-    if (!gModeBaseAliasesInstances.Contains(this))
-    {
-        gModeBaseAliasesInstances.PushBack(this);
-    }
-}
-
-void Instance::UnregisterInstance()
-{
-    gModeBaseAliasesInstances.Remove(this);
-}
-
-bool Instance::HasFeature(Feature feature) const
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+    CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+    AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mDelegate(aDelegate), mEndpointId(aEndpointId),
+    mClusterId(aClusterId),
+    mCurrentMode(0), // This is a temporary value and may not be valid. We will change this to the value of the first
+    // mode in the list at the start of the Init function to ensure that it represents a valid mode.
```

Comment:
This comment got mis-indented and a lot less readable as part of the move.

Reply:
Fixed.